### ISSUE TITLE
Passing ACME http-01 challenge requests to services stopped working v3.0

### ISSUE THREAD

**Issue by Feelemoon** [traefik/traefik#10861]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

Some services behind traefik want to run their own ACME client. This worked fine with traefik v2.x, but isn't working anymore with v3.0. The ACME http challenge request is always intercepted by traefik, regardless of the router priority. Even if a matching challenge isn't found, the request won't be passed downstream. I've increased it to MaxInt - 1000, but this didn't change anything.

### What did you see instead?

Increasing the router priority does not work anymore. This seems to be related to this commit:
https://github.com/traefik/traefik/commit/1e7dbc70a0a28f4ad7bed92d845faecdf2cb97d9

### What version of Traefik are you using?

From `docker inspect`:

```
                "org.opencontainers.image.description": "A modern reverse-proxy",
                "org.opencontainers.image.documentation": "https://docs.traefik.io",
                "org.opencontainers.image.source": "https://github.com/traefik/traefik",
                "org.opencontainers.image.title": "Traefik",
                "org.opencontainers.image.url": "https://traefik.io",
                "org.opencontainers.image.vendor": "Traefik Labs",
                "org.opencontainers.image.version": "v3.0.3"
```

### What is your environment & configuration?

```yaml
http:
  router:
    acmeRouter:
      entryPoints: ["web"]
      rule: "Host(`acme.example.com`) && PathPrefix(`/.well-known/acme-challenge/`)"
      middlewares:
        - serverHeader
      priority: 4294967296
      service: acmeService
```

Add more configuration information here.

### If applicable, please paste the log output in DEBUG level

```
2024-06-28T08:00:08Z DBG github.com/traefik/traefik/v3/pkg/provider/acme/challenge_http.go:104 > Retrieving the ACME challenge for acme.example.com (token "tCPN5wxK_3tUNtJZHFlulA")... providerName=acme
2024-06-28T08:00:08Z ERR github.com/traefik/traefik/v3/pkg/provider/acme/challenge_http.go:110 > Cannot retrieve the ACME challenge for acme.example.com (token "tCPN5wxK_3tUNtJZHFlulA") providerName=acme
```

**Comment by sebadob** [comment]
Most probably related to https://github.com/traefik/traefik/issues/10702

**Comment by mmatur** [comment]
Thanks for your issue @Feelemoon 

Seems a duplicate of https://github.com/traefik/traefik/issues/10702.

Feel free to reopen-it if necessary.

**Comment by Feelemoon** [comment]
@mmatur: #10702 seems to have another cause. Please re-open this issue

**Comment by mmatur** [comment]
Hi @Feelemoon,

Thanks for reporting this.

Before re-opening the issue, could you please provide us a reproductible use case (with your static and dynamic configuration)? It will allow us to determine if it same cause.

**Comment by Feelemoon** [comment]
Hi @mmatur ,

Unfortunately I cannot cut out a simple example from our configuration here. Assume, that there is a web server behind traefik, that itself needs it's own certificate for client authentication.

I think, that you can setup an example with traefik in front of a `certbot/certbot` container as a service. Use the acmeRouter configuration snippet from above to try to redirect the ACME http challenge to certbot. traefik provides an https and an http endpoint and is configured to request certificates from "Let's encrypt" or "ZeroSSL". After traefik has received a certificate for TLS termination, execute certbot in the container to request a certificate with `--standalone`.

**Comment by jgfoster** [comment]
I'm not sure if this is the OP's issue, but here is what I've found:

I have Traefik as a reverse proxy for `foo.example.com` and `bar.example.com`. Traefik is configured to manage TLS for `foo` using LetsEncrypt and everything works fine. The `bar` server used to have it's own public IP address and is configured to use `certbot` to obtain certificates. I'd like to change as little as possible on `bar` (leaving `certbot` to manage certificates), so I configured Traefik to forward `http://bar.example.com/.well-known` to `bar` on port 80 and to forward TCP traffic from 443 on to `bar` on 443 (not using HTTPS because then Traefik will try to terminate the SSL traffic). 

When I do `curl http://bar.example.com/.well-known/no_such_file.txt` it (properly) returns a 404 from `bar` (the Traefik logs says that it forwarded it on to `bar` and the web server on `bar` shows the request in its `access.log`). When I do `curl http://bar.example.com/.well-known/acme-challenge/abcd` I do not get any output from `curl` and the web server on `bar` does not report receiving a request. The Traefik log shows "Cannot retrieve the ACME challenge for bar.example.com ... providerName=acme". 

I've looked at #10702 but am not using Kubernetics or `IngressRoute` (at least not that I know of!).
